### PR TITLE
Update to enable Rubocop and contains Rubocop fixes in lib/

### PR DESCRIPTION
**Pull request by HelenCampbell**

### CODE REVIEW

File: lib/puppet/provider/mysql_datadir/mysql.rb [hunk 1]
```diff
@@ -1,71 +1,71 @@
...
+      debug("Installing MySQL data directory with mysql_install_db #{opts.compact.join(' ')}")
+      mysql_install_db(opts.compact)
     end
 
-   exists?
+    exists?
   end
 
   def destroy
+    # rubocop:disable Lint/UselessAssignment
```
Comment (tphoney) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: this can just go on the offending line

File: lib/puppet/provider/mysql_grant/mysql.rb [hunk 2]
```diff
@@ -1,79 +1,78 @@
...
       end
     end
-    return instances
+    instances
   end
 
   def self.prefetch(resources)
     users = instances
     resources.keys.each do |name|
+      # rubocop:disable Lint/AssignmentInCondition
```
Comment (tphoney) [inline lib/puppet/provider/mysql_grant/mysql.rb]: put on the offending line

File: lib/puppet/provider/mysql_plugin/mysql.rb [hunk 3]
```diff
@@ -19,19 +18,21 @@ def self.instances
   def self.prefetch(resources)
     plugins = instances
     resources.keys.each do |plugin|
+      # rubocop:disable Lint/AssignmentInCondition
```
Comment (tphoney) [inline lib/puppet/provider/mysql_plugin/mysql.rb]: put on the offending line

File: spec/classes/graceful_failures_spec.rb [hunk 4]
```diff
@@ -2,14 +2,14 @@
 
 describe 'mysql::server' do
   context "on an unsupported OS" do
-
-    let(:facts) do
-      {
-        :osfamily => 'UNSUPPORTED',
+
```
Comment (tphoney) [inline spec/classes/graceful_failures_spec.rb]: this looks weird, a misedit ?

File: lib/puppet/parser/functions/mysql_deepmerge.rb [hunk 5]
```diff
@@ -32,25 +33,25 @@ module Puppet::Parser::Functions
...
 end
 
-def has_normalized!(hash, key)
-  return true if hash.has_key?( key )
-  return false unless key.match(/-|_/)
-  other_key = key.include?('-') ? key.gsub( '-', '_' ) : key.gsub( '_', '-' )
-  return false unless hash.has_key?( other_key )
-  hash[key] = hash.delete( other_key )
-  return true;
+def normalized!(hash, key)
```
Comment (eputnam) [inline lib/puppet/parser/functions/mysql_deepmerge.rb]: `normalized?(hash,key)`?

File: lib/puppet/provider/mysql.rb [hunk 6]
```diff
@@ -77,39 +75,33 @@ def self.cmd_table(table)
...
-    end
+    table_string << if table == '*.*'
+                      '*.*'
+                    # Special case also for PROCEDURES
+                    elsif table.start_with?('PROCEDURE ')
+                      table.sub(%r{^PROCEDURE (.*)(\..*)}, 'PROCEDURE `\1`\2')
+                    else
+                      table.sub(%r{^(.*)(\..*)}, '`\1`\2')
+                    end
     table_string
```
Comment (eputnam) [inline lib/puppet/provider/mysql.rb]: do we need this if we're changing to the rubocop'd syntax?

File: lib/puppet/provider/mysql_datadir/mysql.rb [hunk 7]
```diff
@@ -1,71 +1,67 @@
...
-  commands :mysqld => 'mysqld'
-  commands :mysql_install_db => 'mysql_install_db'
+  ENV['PATH'] = ENV['PATH'] + ':/usr/libexec:/usr/share/mysql/scripts:/opt/rh/mysql55/root/usr/bin:/opt/rh/mysql55/root/usr/libexec:/usr/mysql/5.5/bin:/usr/mysql/5.6/bin:/usr/mysql/5.7/bin'
 
+  commands mysqld: 'mysqld'
+  commands mysql_install_db: 'mysql_install_db'
+  # rubocop:disable Lint/UselessAssignment
   def create
-    name                     = @resource[:name]
+    name = @resource[:name]
```
Comment (eputnam) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: this looks kinda goofy, i wonder why it only de-aligned `name`

File: spec/classes/graceful_failures_spec.rb [hunk 8]
```diff
@@ -2,14 +2,12 @@
 
 describe 'mysql::server' do
   context "on an unsupported OS" do
-
     let(:facts) do
-      {
+      {
```
Comment (eputnam) [inline spec/classes/graceful_failures_spec.rb]: lolwut

File: lib/puppet/provider/mysql.rb (hunk 6 above)
Comment (HelenCampbell) [reply lib/puppet/provider/mysql.rb]: Not sure what you mean here? 
Updated with all your other comments 👍

Comment (tphoney) [reply lib/puppet/provider/mysql.rb]: we can fix this later if needs be

**Pull request merged**